### ISSUE TITLE
Errors from mdns-sd in the face of WireGuard interface

### ISSUE THREAD

**Issue by mortenlj** [keepsimple1/mdns-sd#83]
I believe this is a related problem to the one in #52 , only this is on linux and the VPN software is WireGuard.

Everything seems to be working, except an error is logged.
I may be using the library in a bad way (?), because I start many browses for different service types, and every one of those results in the same error being logged, cluttering my logs.

```
ERROR mdns_sd::service_daemon] send to SockAddr { ss_family: 2, len: 16 } via interface Ifv4Addr { ip: 10.255.241.226, netmask: 255.255.248.0, broadcast: None } failed: Required key not available (os error 126)
```

I'm not sure about the best way to deal with this, but to my naive mind, a library shouldn't be logging errors, it should return them to the caller so the caller can decide to log, propagate or ignore.

**Comment by keepsimple1** [comment]
Currently you can use `default-features = false` to disable the logging from this crate. And use `features =<>` to use other features (i.e. `async`) if needed, for example like this: 

```
mdns-sd = { version = "0.5", default-features = false, features = ["async"] }
```

Let me know if that helps. I'm also working on a longer term change that sends log events in a monitoring channel instead of logging directly.

**Comment by mortenlj** [comment]
That helped, but I guess now I won't get any kinds of errors or warnings from mdns?

Anyhow, that's good enough for me for now.

**Comment by keepsimple1** [comment]
Thanks for verifying. I also wanted to understand more about this error.
```
ERROR mdns_sd::service_daemon] send to SockAddr { ss_family: 2, len: 16 } via interface Ifv4Addr { ip: 10.255.241.226, netmask: 255.255.248.0, broadcast: None } failed: Required key not available (os error 126)
```
May I ask couple of questions: 

1. Is the IP `10.255.241.226` the WireGuard interface? 

2. When you create the `ServiceInfo` struct, did you pass in the IPs including the WireGuard interface? or did you let the lib detect the addresses automatically (which is a very recent feature added) ?

**Comment by keepsimple1** [comment]
> That helped, but I guess now I won't get any kinds of errors or warnings from mdns?

I've open a PR to test the socket send when a socket is created. This would help avoiding such errors when the daemon starts to run.

**Comment by mortenlj** [comment]
> What is the `mdns-sd` version you are using?

`mdns-sd = { version = "0.5.10", default-features = false, features = ["async"] }`

> would you have time to help check if the fix work for you? (So you don't have to disable all logging from mdns-sd)

I'm not sure how much time I have for this, this is a low priority, "playing with Rust to scratch a weird itch in my homelab" kind of thing. I can't promise anything, but happy to help if I can.

Btw, the code is pushed to github now: https://github.com/mortenlj/mdns2domeneshop

As you can probably tell, this is just my second attempt at making anything with Rust, and my first attempt doing anything with mdns. I'm about 70% sure I'm doing something very weird, very stupid or both :grimacing:.

**Comment by keepsimple1** [comment]
> I'm not sure how much time I have for this, this is a low priority, "playing with Rust to scratch a weird itch in my homelab" kind of thing. I can't promise anything, but happy to help if I can.

No worries, thanks for letting me know.

**Comment by keepsimple1** [comment]
The fix #84 merged.
